src/components/layout/Sidebar.tsx

```
@@ -1,117 +1,449 @@
+    }
+  }, [user?.role])
+
+  // 🚀 PERFORMANCE: Memoized user clubs with intelligent filtering
+  const userClubs = useMemo(() => {
+    if (!user || !clubs || clubs.length === 0) {
+      return []
+    }
+
+    const filteredClubs = clubs.filter(club => {
+      const isMember = club.memberIds && club.memberIds.includes(user.id)
+      const isLeader = club.leaderId === user.id
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)